### PR TITLE
Fix PassiveSTS failure due to UserPrincipalName values with tenant domain separator

### LINKED ISSUE #4282: PassiveSTS fails for a tenant user with UserPrincipalName claim
**Steps to Reproduce**

- Create a tenant(ex: mywso2.org)
- Create a claim http://wso2.org/claims/userPrincipalName
- Create a tenant user (ex: user) with email as userPricipal name 
- Setup PassiveSTS SP for the tenant
-  In the claim configuration of the Service Provider, add a Custom Claim Dialect with "Service Provider Claim" as http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name and "Local Claim" as http://wso2.org/claims/userPrincipalName 
- Configure the "Subject Claim URI" as http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name
- Check the Requested Claim option for the "Service Provider Claim"

**SAML response:** 
```
<soapenv:Value xmlns:sts="http://wso2.org/passivests">sts:InvalidRequest</soapenv:Value></soapenv:Subcode></soapenv:Code><soapenv:Reason><soapenv:Text>Error in creating a SAMLToken using Opensaml library</soapenv:Text></soapenv:Reason><soapenv:Detail>none</soapenv:Detail> 
```

**IS logs:** 
```
TID: [286] [] [2018-12-18 14:42:33,665] @mywso2.org [286] [IS] WARN {org.wso2.carbon.core.util.AnonymousSessionUtil} - Failed to retrieve Realm for the Invalid Domain : wso2.org
```

### PR DESCRIPTION
**Pull request by ashensw**
### Proposed changes in this pull request

- Resolves wso2/product-is#4282

### When should this PR be merged

- Depends on https://github.com/wso2-extensions/identity-inbound-auth-sts/pull/71

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java [hunk 1]
```diff
@@ -136,9 +147,15 @@ public void handle(SAMLCallback callback) throws SAMLException {
 
                             if (StringUtils.isNotBlank(remoteClaimSuffixValue) &&
                                     StringUtils.isNotBlank(remoteClaimPrefixValue)) {
-                                localClaimValue = IdentityProviderServiceComponent.getRealmService().
-                                        getBootstrapRealm().getUserStoreManager().
-                                        getUserClaimValue(userIdentifier, localClaimUri, "default");
+
+                                if(!authenticatedUser.isFederatedUser()) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Loading claim values from local UserStore");
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java]: can we add the username here

File: components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java [hunk 2]
```diff
@@ -188,12 +205,12 @@ private Attribute getSAML2Attribute(String name, String value, String namespace)
...
      *
      * @param claimsElement
-     * @param userIdentifier
+     * @param tenantDomain
      * @throws IdentityProviderException
      */
-    private void loadClaims(OMElement claimsElement, String userIdentifier) throws IdentityProviderException {
-        IdentityClaimManager claimManager = null;
-        Claim[] claims = null;
+    private void loadClaims(OMElement claimsElement, String tenantDomain) throws IdentityProviderException {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java]: tenantDomain --> spTenantDomain

**Review by mefarazath — APPROVED** [review]

File: components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java (hunk 2 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java]: updated the PR with the changes.

File: components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java (hunk 1 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java]: updated the PR with the changes.

**Review by mefarazath — APPROVED** [review]

**Pull request merged**